servicebus/data-plane/azure-servicebus/src/main/java/com/microsoft/azure/servicebus/primitives/AsyncUtil.java

```
@@ -6,53 +6,51 @@
         } catch (InterruptedException | ExecutionException e) {
             e.printStackTrace();
             return false;
         }
     }
 
-    public static <T> void completeFuture(CompletableFuture<T> future, T result)
-    {
-        MessagingFactory.INTERNAL_THREAD_POOL.submit(new CompleteCallable<T>(future, result));
+    public static <T> void completeFuture(CompletableFuture<T> future, T result) {
+        MessagingFactory.INTERNAL_THREAD_POOL.submit(new CompleteCallable<>(future, result));
     }
 
-    public static <T> boolean completeFutureExceptionallyAndGetStatus(CompletableFuture<T> future, Throwable exception)
-    {
+    public static <T> boolean completeFutureExceptionallyAndGetStatus(CompletableFuture<T> future, Throwable exception) {
         try {
-            return MessagingFactory.INTERNAL_THREAD_POOL.submit(new CompleteExceptionallyCallable<T>(future, exception)).get();
+            return MessagingFactory.INTERNAL_THREAD_POOL.submit(new CompleteExceptionallyCallable<>(future, exception)).get();
         } catch (InterruptedException | ExecutionException e) {
             e.printStackTrace();
             return false;
         }
     }
 
-    public static <T> void completeFutureExceptionally(CompletableFuture<T> future, Throwable exception)
-    {
-        MessagingFactory.INTERNAL_THREAD_POOL.submit(new CompleteExceptionallyCallable<T>(future, exception));
+    public static <T> void completeFutureExceptionally(CompletableFuture<T> future, Throwable exception) {
+        MessagingFactory.INTERNAL_THREAD_POOL.submit(new CompleteExceptionallyCallable<>(future, exception));
     }
 
-    public static void run(Runnable runnable)
-    {
-        MessagingFactory.INTERNAL_THREAD_POOL.submit(runnable);
+    public static void run(Runnable runnable) {
+        try {
+            MessagingFactory.INTERNAL_THREAD_POOL.submit(runnable);
+        } catch (RejectedExecutionException | NullPointerException e) {
+            e.printStackTrace();
```

> e.printStackTrace(); [](start = 11, length = 21)

what is the point of doing this? just to satisfy the tool. And another problem is it is just eating the exception. We shouldn't do it. I would make an exception and don't make the code ugly.

---

No longer an issue. Will suppress it.